a3p-integration/proposals/z:acceptance/auction.test.js

```diff
@@ -0,0 +1,389 @@
+/**
+ * @file In this file we aim to test auctioneer in an isolated manner. Here's the scenario to test;
+ * - Send 100 ATOMs to gov1 from validator
+ * - For book0, ATOM is collateral, set two types of bids; by price and by percentage, user1 is the bidder
+ * - Deposit 100 ATOMs into book0, gov1 is the depositor
+ * - Wait until placed bids get their payouts
+ * - Make sure the depositor gets correct amounts
+ */
+
+/** @typedef {import('./test-lib/sync-tools.js').RetyrOptions} RetryOptions */
+
+import {
+  addPreexistingOracles,
+  agd,
+  agopsInter,
+  agoric,
+  ATOM_DENOM,
+  CHAINID,
+  executeOffer,
+  getPriceQuote,
+  getUser,
+  GOV1ADDR,
+  GOV3ADDR,
+  pushPrices,
+  USER1ADDR,
+  VALIDATORADDR,
+} from '@agoric/synthetic-chain';
+import '@endo/init';
+import test from 'ava';
+import { boardSlottingMarshaller, makeFromBoard } from './test-lib/rpc.js';
+import {
+  retryUntilCondition,
+  waitUntilAccountFunded,
+  waitUntilOfferResult,
+} from './test-lib/sync-tools.js';
+import { AmountMath } from '@agoric/ertp';
+
+export const scale6 = x => BigInt(Math.round(x * 1_000_000));
+
+const ambientAuthority = {
+  query: agd.query,
+  follow: agoric.follow,
+  setTimeout: globalThis.setTimeout,
+};
+
+const fromBoard = makeFromBoard();
+const marshaller = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+
+export const bankSend = (from, addr, wanted) => {
+  const chain = ['--chain-id', CHAINID];
+  const fromArg = ['--from', from];
+  const testKeyring = ['--keyring-backend', 'test'];
+  const noise = [...fromArg, ...chain, ...testKeyring, '--yes'];
+
+  return agd.tx('bank', 'send', from, addr, wanted, ...noise);
+};
```

Comment:
No, because the implementation from synthetic-chain only uses validator as it `from` address whereas this one `from` is also parameterized. See https://github.com/Agoric/agoric-3-proposals/blob/422b163fecfcf025d53431caebf6d476778b5db3/packages/synthetic-chain/src/lib/agd-lib.ts#L136.

Reply:
My theory about this is that we should improve the tools in synthetic chain as we go, so even though it means doing the work twice, I recommend adding a comment here saying that the changes will be migrated to synthetic-chain, and imported from there once released, and then making a PR for synthetic-chain to add the new parameter.

Reply:
Is this a blocker for this PR? I added a comment saying that we should import `bankSend` from synthetic-chain. See https://github.com/Agoric/agoric-sdk/blob/50a0e17318c083daea6fc2628216adf294525e7a/a3p-integration/proposals/z%3Aacceptance/test-lib/auction-lib.js#L38

Reply:
> Is this a blocker for this PR?

Nope. I expected the two tasks to proceed in parallel.

Reply:
I believe this should be tracked here https://github.com/Agoric/agoric-3-proposals/issues/187